plugins/processors/topk/topk.go

```diff
@@ -0,0 +1,468 @@
+  # drop_no_group = true
+
+  ## Drop the metrics that do not make the cut for the top k
+  # drop_non_top = true          
+
+  ## The plugin assigns each metric a GroupBy tag generated from its name and
+  ## tags. If this setting is different than "" the plugin will add a
+  ## tag (which name will be the value of this setting) to each metric with
+  ## the value of the calculated GroupBy tag. Useful for debugging
+  # add_groupby_tag = ""          
+
+  ## These settings provide a way to know the position of each metric in
+  ## the top k. The 'add_rank_field' setting allows to specify for which
+  ## fields the position is required. If the list is non empty, then a field
+  ## will be added to each every metric for each field present in the 
+  ## 'add_rank_field'. This field will contain the ranking of the group that
+  ## the metric belonged to when aggregated over that field.
+  ## The name of the field will be set to the name of the aggregation field,
+  ## suffixed by the value of the 'rank_field_suffix' setting
+  # add_rank_field = []
+  # rank_field_suffix = "_rank"
+
+  ## These settings provide a way to know what values the plugin is generating
+  ## when aggregating metrics. The 'add_agregate_field' setting allows to
+  ## specify for which fields the final aggregation value is required. If the
+  ## list is non empty, then a field will be added to each every metric for
+  ## each field present in the 'add_aggregate_field'. This field will contain
+  ## the computed aggregation for the group that the metric belonged to when
+  ## aggregated over that field.
+  ## The name of the field will be set to the name of the aggregation field,
+  ## suffixed by the value of the 'aggregate_field_suffix' setting
+  # add_aggregate_field = []
+  # aggregate_field_suffix = "_aggregate"
+`
+
+type MetricAggregation struct {
+	groupbykey string
+	values     map[string]float64
+}
+
+func sortMetrics(metrics []MetricAggregation, field string, reverse bool) {
+	less := func(i, j int) bool {
+		iv := metrics[i].values[field]
+		jv := metrics[j].values[field]
+		if iv < jv {
+			return true
+		} else {
+			return false
+		}
+	}
+
+	if reverse {
+		sort.SliceStable(metrics, less)
+	} else {
+		sort.SliceStable(metrics, func(i, j int) bool { return !less(i, j) })
+	}
+}
+
+func (t *TopK) SampleConfig() string {
+	return sampleConfig
+}
+
+func (t *TopK) Reset() {
+	t.cache = make(map[string][]telegraf.Metric)
+	t.lastAggregation = time.Now()
+}
+
+func (t *TopK) Description() string {
+	return "Print all metrics that pass through this filter."
+}
+
+func (t *TopK) generateGroupByKey(m telegraf.Metric) (string, error) {
+	if t.SimpleTopk {
+		return strconv.FormatUint(m.HashID(), 16), nil
+	}
+
+	// Create the filter.Filter objects if they have not been created
+	if t.tagsGlobs == nil && len(t.GroupBy) > 0 {
+		var err error
+		t.tagsGlobs, err = filter.Compile(t.GroupBy)
+		if err != nil {
+			log.Printf("Could not complile tags globs: %s", t.GroupBy)
```

Comment:
Ok

Reply:
Ps. This is logged in the groupBy function. I applied the change that you suggested, and it will return the error to this block:
```
	// Generate the metric group key
	groupkey, err := t.generateGroupByKey(m)
	if err != nil {
		// If we could not generate the groupkey, fail hard
		// by dropping this and all subsequent metrics
		log.Print("E! [processors.topk]: could not generate group key: %v", err)
		return
	}
```

Which will just log and drop the metrics.